src/codegen/generate.ts

```diff
@@ -1215,57 +1212,121 @@ export default class ApiGenerator {
           return name;
         };
 
-        // build the method signature - first all the required parameters
-        const methodParams = required.map((p) =>
-          cg.createParameter(getArgName(this.resolve(p)), {
-            type: this.getTypeFromParameter(p),
-          }),
-        );
-
-        let body: OpenAPIV3.RequestBodyObject | undefined;
-        let bodyVar;
+        const methodParams: ts.ParameterDeclaration[] = [];
+        let body: OpenAPIV3.RequestBodyObject | undefined = undefined;
+        let bodyVar: string | undefined = undefined;
+        switch (this.opts.argumentStyle ?? "positional") {
+          case "positional":
+            // split into required/optional
+            const [required, optional] = _.partition(parameters, "required");
+
+            // build the method signature - first all the required parameters
+            const requiredParams = required.map((p) =>
+              cg.createParameter(getArgName(this.resolve(p)), {
+                type: this.getTypeFromParameter(p),
+              }),
+            );
+            methodParams.push(...requiredParams);
 
-        // add body if present
-        if (requestBody) {
-          body = this.resolve(requestBody);
-          const schema = this.getSchemaFromContent(body.content);
-          const type = this.getTypeFromSchema(schema, undefined, "writeOnly");
-          bodyVar = toIdentifier(
-            (type as any).name || getReferenceName(schema) || "body",
-          );
-          methodParams.push(
-            cg.createParameter(bodyVar, {
-              type,
-              questionToken: !body.required,
-            }),
-          );
-        }
+            // add body if present
+            if (requestBody) {
+              body = this.resolve(requestBody);
+              const schema = this.getSchemaFromContent(body.content);
+              const type = this.getTypeFromSchema(
+                schema,
+                undefined,
+                "writeOnly",
+              );
+              bodyVar = toIdentifier(
+                (type as any).name || getReferenceName(schema) || "body",
+              );
+              methodParams.push(
+                cg.createParameter(bodyVar, {
+                  type,
+                  questionToken: !body.required,
+                }),
+              );
+            }
```

Comment:
This seems to be duplicated in both switch cases, can we keep remove the duplication?